ceph/rbd/workflows/migration.py

```diff
@@ -0,0 +1,79 @@
+import json
+
+from ceph.rbd.utils import exec_cmd
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def migration_prepare(src_spec, dest_spec, client_node, **kw):
+    """Migration Prepare.
+
+    This method prepare the live migration of image from source to destination
+
+    Args:
+        src_spec : source image spec SOURCE_POOL_NAME/SOURCE_IMAGE_NAME
+                or json formatted string for streamed imports
+        dest_spec: Target image spec TARGET_POOL_NAME/SOURCE_IMAGE_NAME
+    """
+
+    log.info("Starting prepare Live migration of image")
+    src_spec = json.dumps(src_spec)
+    if "{" not in src_spec:
+        command = "rbd migration prepare "
+        command += f"{src_spec} {dest_spec}"
+    else:
+        command = f"echo '{src_spec}' | rbd migration prepare "
+        command += f"{dest_spec} --import-only --source-spec-path -"
+    return exec_cmd(node=client_node, cmd=command)
+
+
+def migration_action(action, dest_spec, client_node, **kw):
+    """Migration action.
+
+    This method will execute/commit the live migration as per the provided action
+
+    Args:
+        action: execute or commit
+        dest_spec: Target image spec in format of TARGET_POOL_NAME/SOURCE_IMAGE_NAME
+    """
+    log.info(f"Starting the {action} migration process")
+    return exec_cmd(
+        node=client_node, cmd=f"rbd migration {action} {dest_spec}", long_running=True
```

Comment:
Why have you not implemented the CLI way for any of these? Its important that we maintain a single standard. We cannot each go about doing our own way of code it will become clumsy.
Please implement the CLI way.

Reply:
Done.